openproblems/tasks/spatial_decomposition/methods/nmfreg.py

```diff
@@ -0,0 +1,106 @@
+from ....tools.decorators import method
+from ....tools.utils import check_version
+from scipy.optimize import nnls
+from scipy.sparse import issparse
+from sklearn.decomposition import NMF
+from sklearn.preprocessing import StandardScaler
+
+import numpy as np
+
+
+@method(
+    method_name="NMF-reg",
+    paper_name="Slide-seq: A scalable technology for measuring genome-wide expression at high spatial resolution",  # noqa: E501
+    paper_url="https://science.sciencemag.org/content/363/6434/1463",  # noqa: E501
+    paper_year=2019,
+    code_url="https://github.com/tudaga/NMFreg_tutorial",
+    code_version=check_version("nmf-reg"),
+)
+def nmfreg(adata):
+    """NMF-reg re-implementation from https://github.com/tudaga/NMFreg_tutorial."""
```

Comment:
Some more commenting on this would be helpful

Reply:
added one line